### ISSUE TITLE
Uploading to Genuis controller with RGBW 4-channel string does not set Type correctly.

### ISSUE THREAD

**Issue by nhutchison** [xLightsSequencer/xLights#3539]
No sequence is required to duplicate this issue.

After creating a single pixel floodlight, using a 4 channel WRGB String type in the layout, and assigning the model to a port in the visualizer, when uploading to the controller, the type is set to RGB, and not RGBW.

To reproduce, follow the steps below:

1. Create a single line model
2. Set the # pixels to 1
3. Under String properties, set the String Type to 4 Channel WRGB
4. Save the model
5. Under the Genuis controller, open the visualizer, and assign the new model to a free port.
6. Note that the string is not sharing a port in this replication
7. Close the visualizer
8. Clock Upload Outputs
9. On the Genuis controller, check the outputs.  Note the Type should be RGBW.

Xlights version used is 2022.21.2 on a Mac OS 11.6
Genius controller is running v1.0.6 FW

**Comment by grumblegeek** [comment]
I recently purchased some RGBW floods and am experiencing the same issue.  Anytime I upload outputs to my Genius I have to remember to go to the outputs in my Genius and change it to RBGW in order to get the correct output.

**Comment by computergeek1507** [comment]
What String Type do you have configured for the model in xLights?

**Comment by computergeek1507** [comment]
You need to set the color order in the Controller Connection section of the Model Properties


![image](https://github.com/smeighan/xLights/assets/2293065/0c7c1872-4e29-4eae-a662-ffd586e0c907)

**Comment by grumblegeek** [comment]
That work!  Thanks.  I had not noticed that setting before.

**Comment by keithsw1111** [comment]
@computergeek1507 that should not be necessary unless you want to change from the default. The string properties already tell you these are 4 channel pixels ... controller connection should just set any reordering setting available on the controller.

**Comment by computergeek1507** [comment]
@keithsw1111 All the controller upload code except FPP don't look at the model string type during upload, they use the color order override or use the default "RGB". 

![image](https://github.com/smeighan/xLights/assets/2293065/9da1b9b7-1c49-4201-8a31-ab0d9ce332fa)

**Comment by keithsw1111** [comment]
Not true ... that line of code is for controller based colour ordering. The node counting code is 

![image](https://github.com/smeighan/xLights/assets/10283944/4de5976a-5b5a-4421-9411-66f37fa11105)

If the protocol is 3 channel but the channels are a multiple of 4 because the string type is a 4 channel per node then this increases the 3 channel node count to compensate. It does not need the colour order to be set.

**Comment by computergeek1507** [comment]
The issue is specifying that if the Model String Type is set to 4 Channel WRGB, it doesn't change the "Node Type". The "model->GetStringType()" is not referenced in the the any controller upload besides FPP. 
![image](https://github.com/smeighan/xLights/assets/2293065/3172a6b4-3661-46b5-9467-b4ff6ec9601d)

The "GetChannelsPerPixel" code looks at the controller pixel protocol selected not "Model String Type", so Yes if you select a protocal of "sk6812rgbw" it will changed the pixel count but not if you selected Model String Type of 4 Channel WRGB as the issues states. 

![image](https://github.com/smeighan/xLights/assets/2293065/cb6c660a-b2a4-4ae8-b4d9-15e96177a5a0)

I'm still confused what is incorrect with this upload code, the Genius controllers support 4 channel pixels and will adjust it's 
underlying channel buffer accordingly. xLights doesn't need to add extra pixels to accommodate the extra channels. I don't understand what else is missing.

**Comment by dkulp** [comment]
I agree with Scott, if the controller natively supports 4 channel pixels, xLights should NOT be adjusting the pixel count to compensate.   That's just confusing for the user.   For controllers that don't have native 4 channel support, then sure, but the modern controllers should set the number of pixels to the actual number of pixels and make sure the pixel type is set appropriately.

**Comment by keithsw1111** [comment]
Your call ... your controllers ... but users have been known to want to mix 3 and 4 channel pixels with compatible signals on the same port and this approach makes this unnecessarily hard.

**Comment by dkulp** [comment]
Not for controllers that support virtual strings where each virtual string can be setup as either 3 channel or 4 channel.

**Comment by keithsw1111** [comment]
True ... but I dont think the genius do ... and that is what this issue was originally about.

**Comment by derwin12** [comment]
I don't see any work left to do here..
